src/index.js

```diff
@@ -21,12 +24,19 @@ app.use(async (ctx, next) => {
   }
 });
 
+app.use(session({}, app));
+app.use(passport.initialize());
+app.use(passport.session());
```

Comment:
Actually we don't need to store "user" in session. We are not authenticating user anyway, we just get a token in a one-time OAuth2 process and then store it to database.

`passport.session()`, `passport.serializeUser` and `passport.deserializeUser` are used to store "user" in session. I think after removing these together, LINE notify would still work?

Reply:
According to [doc](https://notify-bot.line.me/doc/en/), session is used to avoid CSRF.
`Assigns a token that can be used for responding to CSRF attacksCSRF attacks are typically countered by assigning a hash value generated from a user's session ID, and then verifying the state parameter variable when it attempts to access redirect_uri.` 

I've tried removing `passport.serializeUser` and `passport.deserializeUser`, there would be errors occur. But I didn't investigate deeper.

Reply:
Interesting aspect from LINE notify doc. The doc is saying that we can pass a `state` in URL param. OAuth2 service will return the `state` URL param intact, so that our callback can verify the `state`.

It seems that it can be achieved by passing `state: true` to the strategy constructor: https://github.com/jaredhanson/passport-oauth2/pull/93/files

As far as I know, W what passport.srssion() does should be serializing user from session to `ctx.state.user`, and vice versa. It's probably not related to the anti-CSRF measure with `state` param ad mentioned in LINE Notify doc.